File: examples/developer-hub-python/chain_id_flare.py
import asyncio

from web3 import AsyncHTTPProvider, AsyncWeb3


async def main() -> int:
    w3 = AsyncWeb3(
        AsyncHTTPProvider("https://flare-api.flare.network/ext/C/rpc"),
    )
    chain_id = await w3.eth.chain_id
    print(f"Chain ID: {chain_id}")
    # Chain ID: 14
    return chain_id


if __name__ == "__main__":
    asyncio.run(main())

File: examples/developer-hub-rust/src/bin/chain_id_flare.rs
use alloy::providers::{Provider, ProviderBuilder};
use eyre::Result;

#[tokio::main]
async fn main() -> Result<()> {
    let provider =
        ProviderBuilder::new().on_http("https://flare-api.flare.network/ext/C/rpc".parse()?);
    let chain_id = provider.get_chain_id().await?;
    println!("Chain ID: {}", chain_id); // Chain ID: 14
    Ok(())
}
